src/Compute/Compute/Extension/AzureDiskEncryption/SetAzureDiskEncryptionExtension.cs

```diff
@@ -541,9 +541,22 @@ public override void ExecuteCmdlet()
 
                     currentOSType = virtualMachineResponse.StorageProfile.OsDisk.OsType;
 
+                    var vmParameters = (this.ComputeClient.ComputeManagementClient.VirtualMachines.Get(
+                        this.ResourceGroupName, VMName));
+
                     if (OperatingSystemTypes.Linux.Equals(currentOSType) && !SkipVmBackup)
                     {
-                        CreateVMBackupForLinx();
+                        if (vmParameters.StorageProfile.OsDisk.ManagedDisk != null)
+                        {
+                            ThrowTerminatingError(new ErrorRecord(new ArgumentException(string.Format(CultureInfo.CurrentUICulture, "-skipVmBackup parameter is a required parameter for encrypting Linux VMs with managed disks. For more information, see https://docs.microsoft.com/azure/security/azure-security-disk-encryption-linux.")),
```

Comment:
We are trying to have all strings taken from the Resources.resx file. Could you do it?

Reply:
I am trying to throw an error in the powershell window and not have a pop-up window display the error. Existing usages of Resources.resx file seem to be having a pop-up displaying the message.  Should I still be using Resources.resx?

Reply:
Resources.resx would just hold the string for future localization

Reply:
@niander I have added changes. Can you please take a look?